src/lib/components/themeswitcher.svelte

```diff
@@ -0,0 +1,44 @@
+<script lang="ts">
+  import { onMount } from 'svelte';
+
+  let themes = ['light', 'dark'];
+  let currentTheme: string = 'light';
+
+  // Function to set the theme
+  function setTheme(theme: string) {
+    currentTheme = theme;
+    document.documentElement.setAttribute('data-theme', theme);
+    localStorage.setItem('theme', theme);
+  }
+
+  // automatically set the theme based on the user's preference
+  onMount(() => {
+    const storedTheme = localStorage.getItem('theme');
+    if (storedTheme) {
+      setTheme(storedTheme);
+    } else if (window.matchMedia('(prefers-color-scheme: dark)').matches) {
+      setTheme('dark');
+    } else {
+      setTheme('light');
+    }
+  });
+
+  function handleSetTheme(event: any) {
+    const theme = event.target.value;
+    setTheme(theme);
+  }
+</script>
+
+<div class={$$props.class}>
+  <div class="flex flex-row items-center">
```

Comment:
Braucht es diese zwei ineinander verschachtelten DIV's oder kann man auch einen nehmen? Und ist das `class={$$props.class}` notwendig oder nur 'falls man das mal braucht'?

Reply:
class={$$props.class} ist dafür, dass man dem parent container die styles geben kann. ich würde gerne die komponente immer auf 100% breite konzipieren und die breiteinschränkung bzw. das layouting durch den parent (außerhalb der component, die ja ggf. auch an anderer stelle verwendet werden könnte)

Reply:
ich denke man braucht die verschachtelung nicht in jedem fall, aber es bietet mehr flexibilität, vor allem wenn es um paddings geht die meine ich falsch berechnet werden auf flex containern, kann aber sein dass ich mich täusche und das war nur bei den flex-items so dass man kein padding geben darf